### ISSUE TITLE
Mod key is never released

### ISSUE THREAD

**Issue by yangrq1018** [hyprwm/Hyprland#4378]
### Hyprland Version

<details> <summary>System/Version info</summary>

```sh
Hyprland, built from branch main at commit 7e033e48ace5406a9bc442f7d403f9ce3af193f3 dirty (make: unbreak with non-GNU ln(1) after 78f9ba9fdd7e).
Date: Sat Jan 6 01:22:43 2024
Tag: v0.34.0-25-g7e033e48

flags: (if any)


System Information:
System name: Linux
Node name: martin-XiaoXinPro-14-IAH5R
Release: 6.2.0-39-generic
Version: #40~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Thu Nov 16 10:53:04 UTC 2


GPU information:
00:02.0 VGA compatible controller [0300]: Intel Corporation Alder Lake-P Integrated Graphics Controller [8086:46a6] (rev 0c) (prog-if 00 [VGA controller])


os-release: PRETTY_NAME="Ubuntu 22.04.3 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.3 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy


plugins:
```

</details>

### Bug or Regression?

Bug

### Description

I have tried to set my primary mod key to SUPER and ALT_L, both keys have experienced this problem: in some cases even if I have released the mod key, hyperland still thinks it's pressed. This continues for a while, perhaps after some clicking,  some workspace switching, the glitch is gone.

### How to reproduce

This happens a lot on browsers, and much less on terminal emulators. It seems even if the mod key is "jammed", typing in terminal emulator is still fine (I could still type C, and hyperland won't close the window as mod + C).

### Crash reports, logs, images, videos

Sorry I have checked the /tmp log, they don't have much useful info, hyperland doesn't crash on this bug. Perhaps I need some keyboard event logs, but I am quite new to hyperland, I would be happy to provide those if instructed.

PS: since I am on Ubuntu 22.04, I pretty much built everything from source. Other aspects of hyperland work fine on
my PC so far, it's just this bug a bit annoying.

**Comment by yangrq1018** [comment]
Could be a libxkbcommon outdated issue. I installed from apt on Ubuntu 22.04. Have compiled from source to see. Closed for now.

**Comment by yangrq1018** [comment]
[hyprland.log](https://github.com/hyprwm/Hyprland/files/13864751/hyprland.log)
Log attached.
"not found in m_pressedKeys" looks suspicious.

**Comment by vaxerski** [comment]
looks like regressed by #4304 @rszyma ?

**Comment by rszyma** [comment]
I'll look into it

**Comment by rszyma** [comment]
Another one:
1. Set keybind to `bind=SUPER ALT_L, ALT_L, exec, test`
2. Press `SUPER_L`
3. Release `SUPER_L`
4. `SUPER_L` remains pressed.

Same with `r` flag.

**Comment by rszyma** [comment]
Fix 1 bug create 3 another. Classic

**Comment by vaxerski** [comment]
any chance of a fix? :)

**Comment by rszyma** [comment]
yep, I'll try

**Comment by yangrq1018** [comment]
I tried to simply ignore the "not found in m_pressedKeys" case, by this
```cpp
diff --git a/src/managers/KeybindManager.cpp b/src/managers/KeybindManager.cpp
index 21cc1f74..0f2f032e 100644
--- a/src/managers/KeybindManager.cpp
+++ b/src/managers/KeybindManager.cpp
@@ -354,6 +354,8 @@ bool CKeybindManager::onKeyEvent(wlr_keyboard_key_event* e, SKeyboard* pKeyboard
             Debug::log(ERR, "BUG THIS: key not found in m_dPressedKeys");
             // fallback with wrong `KEY.modmaskAtPressTime`, this can be buggy
             found = handleKeybinds(MODS, KEY, false);
+            m_dPressedKeys.clear();
+            return true;
         }

         shadowKeybinds();

```
Kind of work around the bug, no jammed keys so far.

**Comment by yangrq1018** [comment]
I switched to Arch and the pacman binary 0.34.0 has no problem at all.

**Comment by rszyma** [comment]
> I switched to Arch and the pacman binary 0.34.0 has no problem at all.

Perhaps you hit a different bug (from above I showcased) with Hyprland, that is somehow only specific to ubuntu? In the reproduction for the bugs I showcased above, when I recollect again, I haven't had "not found in m_pressedKeys" logged, so this seems even more likely.

**Comment by vaxerski** [comment]
fixed rszyma's issue in d484506600e8e1065bef8df0ec0ed61393601a4d, check if yours is too.

**Comment by yangrq1018** [comment]
I will check, but I don't have an ubuntu environment now. I will try build from souce on arch later.

**Comment by rszyma** [comment]
> fixed rszyma's issue in [d484506](https://github.com/hyprwm/Hyprland/commit/d484506600e8e1065bef8df0ec0ed61393601a4d)

nice, thank you. 

___

There's one more thing that I noticed that don't seem to work properly:
```
bindr = , ALT_L, exec, test
``` 
Expected: exec on release
Actual: nothing 

A workaround is to use this instead:
```
bindr = ALT, ALT_L, exec, test
```

**Comment by vaxerski** [comment]
wiki does mention that, though.
